tests/components/canary/test_init.py

```diff
@@ -1,37 +1,82 @@
 
-from tests.async_mock import patch
+from . import YAML_CONFIG, init_integration
 
+from tests.async_mock import patch
 
-async def test_setup_with_valid_config(hass, canary) -> None:
-    """Test setup with valid YAML."""
-    await async_setup_component(hass, "persistent_notification", {})
-    config = {DOMAIN: {"username": "test-username", "password": "test-password"}}
 
+async def test_import_from_yaml(hass, canary) -> None:
+    """Test import from YAML."""
     with patch(
-        "homeassistant.components.canary.alarm_control_panel.setup_platform",
+        "homeassistant.components.canary.async_setup_entry",
         return_value=True,
-    ), patch(
-        "homeassistant.components.canary.camera.setup_platform",
-        return_value=True,
-    ), patch(
-        "homeassistant.components.canary.sensor.setup_platform",
+    ):
+        assert await async_setup_component(hass, DOMAIN, {DOMAIN: YAML_CONFIG})
+        await hass.async_block_till_done()
+
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(entries) == 1
+
+    assert entries[0].data[CONF_USERNAME] == "test-username"
+    assert entries[0].data[CONF_PASSWORD] == "test-password"
+    assert entries[0].data[CONF_TIMEOUT] == 5
+
+
+async def test_import_from_yaml_ffmpeg(hass, canary) -> None:
+    """Test import from YAML with ffmpeg arguments."""
+    with patch(
+        "homeassistant.components.canary.async_setup_entry",
         return_value=True,
     ):
-        assert await async_setup_component(hass, DOMAIN, config)
+        assert await async_setup_component(
+            hass,
+            DOMAIN,
+            {
+                DOMAIN: YAML_CONFIG,
+                CAMERA_DOMAIN: [{"platform": DOMAIN, CONF_FFMPEG_ARGUMENTS: "-v"}],
+            },
+        )
         await hass.async_block_till_done()
 
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(entries) == 1
+
+    assert entries[0].data[CONF_USERNAME] == "test-username"
+    assert entries[0].data[CONF_PASSWORD] == "test-password"
+    assert entries[0].data[CONF_TIMEOUT] == 5
+    assert entries[0].data.get(CONF_FFMPEG_ARGUMENTS) == "-v"
+
 
-async def test_setup_with_http_error(hass, canary) -> None:
-    """Test setup with HTTP error."""
-    await async_setup_component(hass, "persistent_notification", {})
-    config = {DOMAIN: {"username": "test-username", "password": "test-password"}}
+async def test_unload_entry(hass, canary):
+    """Test successful unload of entry."""
+    entry = await init_integration(hass)
 
-    canary.side_effect = HTTPError()
+    assert entry
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert entry.state == ENTRY_STATE_LOADED
 
-    assert not await async_setup_component(hass, DOMAIN, config)
+    assert await hass.config_entries.async_unload(entry.entry_id)
     await hass.async_block_till_done()
+
+    assert entry.state == ENTRY_STATE_NOT_LOADED
+    assert not hass.data.get(DOMAIN)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)